de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/data/GitAccess.java

```diff
@@ -0,0 +1,155 @@
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.NoHeadException;
+import org.eclipse.jgit.api.errors.NoMessageException;
+import org.eclipse.jgit.api.errors.UnmergedPathsException;
+import org.eclipse.jgit.api.errors.WrongRepositoryStateException;
+
+public class GitAccess {
+
+	private static GitAccess gitAccess = null;
+
+	private GitAccess() {
+	}
+
+	public static GitAccess getInstance() {
+		if (gitAccess == null) {
+			gitAccess = new GitAccess();
+		}
+
+		return gitAccess;
+	}
+
+	/**
+	 * Clones a git repository from the given uri to a local repository.
+	 * @param uri git uri to clone from
+	 * @param localRoot directory where the cloned repository should be put below
+	 * @return local directory where the git repository was cloned to, if successful; error message otherwise
+	 */
+	public String cloneRepository(String uri, String localRoot) {
+		String directory = extractDirectoryFromUri(uri);
+		directory = FilenameUtils.concat(localRoot, directory);
+		String result = directory;
+		
+		try {
+			Git.cloneRepository()
+				.setURI(uri)
+				.setDirectory(new File(directory))
+				.call();
+		} catch (GitAPIException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		}
+
+		return result;
+	}
+
+	/**
+	 * Commits the changes of the local directory of the repository belonging to the given URI.
+	 * Three commands are executed: add all files, commit (with message), push.
+	 * @param localDirectory directory of local repository
+	 * @param message commit message
+	 * @return "ok" if everything was ok; exception message if there was an exception
+	 */
+	public String commit(String localDirectory, String message) {
+		String result = "ok";
+
+		try {
+			Git git = Git.open(new File(localDirectory));
+			git.add()
+				.addFilepattern(".")
+				.call();
+			git.commit()
+				.setAll(true)
+				.setMessage(message)
+				.call();
+			git.push()
+				.call();
+		} catch (NoHeadException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		} catch (NoMessageException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		} catch (UnmergedPathsException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		} catch (ConcurrentRefUpdateException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		} catch (WrongRepositoryStateException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		} catch (AbortedByHookException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		} catch (GitAPIException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		} catch (IOException e) {
+			result = e.getMessage();
+			e.printStackTrace();
+		}
+		return result;
+	}
+
+	/**
+	 * 
+	 * @param localDirectory directory of local repository
+	 * @return
+	 */
+	public String update(String localDirectory) {
+		String result = "ok";
+		Git git;
+		try {
+			git = Git.open(new File(localDirectory));
+			git.pull()
+				.call();

```

Comment:
fix indentation

Reply:
no, it is the same pattern as for all the other calls. In this case it should be ok.